src/query/execute_schema.c

```diff
@@ -10016,6 +10020,366 @@ do_alter_change_tbl_comment (PARSER_CONTEXT * const parser, PT_NODE * const alte
+      if (error != NO_ERROR)
+       {
+	 goto exit;
+       }
+    }
+  else
+    {
+      assert (change_mode == SM_ATTR_CHG_ONLY_SCHEMA);
+    }
+
+  /* create any new constraints: */
+  if (attr_chg_prop.new_constr_info != NULL)
+    {
+      SM_CONSTRAINT_INFO *ci = NULL;
+
+      error = sort_constr_info_list (&(attr_chg_prop.new_constr_info));
+      if (error != NO_ERROR)
+       {
+	 er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_UNEXPECTED, 1, "Sorting constraints failed.");
+	 goto exit;
+       }
+
+      /* add new constraints */
+      for (ci = attr_chg_prop.new_constr_info; ci != NULL; ci = ci->next)
+       {
+	 if (ci->constraint_type == DB_CONSTRAINT_NOT_NULL)
+	   {
+	     const char *att_name = *(ci->att_names);
+
+	     if (alter->info.alter.hint & PT_HINT_SKIP_UPDATE_NULL)
+	       {
+		 error = db_add_constraint (class_mop, ci->constraint_type, NULL, (const char **) ci->att_names, 0);
+	       }
+	     else if (!prm_get_bool_value (PRM_ID_ALTER_TABLE_CHANGE_TYPE_STRICT))
+	       {
+		 char query[SM_MAX_IDENTIFIER_LENGTH * 4 + 36] = { 0 };
+		 const char *class_name = NULL;
+		 const char *hard_default =
+		   get_hard_default_for_type (alter->info.alter.alter_clause.attr_mthd.attr_def_list->type_enum);
+		 int update_rows_count = 0;
+
+		 class_name = db_get_class_name (class_mop);
+		 if (class_name == NULL)
+		   {
+		     error = ER_UNEXPECTED;
+		     er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, error, 1, "Cannot get class name of mop.");
+		     goto exit;
+		   }
+
+		 assert (class_name != NULL && att_name != NULL && hard_default != NULL);
+
+		 snprintf (query, SM_MAX_IDENTIFIER_LENGTH * 4 + 30, "UPDATE [%s] SET [%s]=%s WHERE [%s] IS NULL",
+			   class_name, att_name, hard_default, att_name);
+		 error = do_run_update_query_for_class (query, class_mop, &update_rows_count);
+		 if (error != NO_ERROR)
+		   {
+		     goto exit;
+		   }
+
+		 if (update_rows_count > 0)
+		   {
+		     er_set (ER_WARNING_SEVERITY, ARG_FILE_LINE, ER_ALTER_CHANGE_ADD_NOT_NULL_SET_HARD_DEFAULT, 0);
+		   }
+		 error = db_add_constraint (class_mop, ci->constraint_type, NULL, (const char **) ci->att_names, 0);
+	       }
+	     else
+	       {
+		 error = db_constrain_non_null (class_mop, *(ci->att_names), 0, 1);
+	       }
+	     if (error != NO_ERROR)
+	       {
+		 goto exit;
+	       }
+	   }
+	 else
+	   {
+	     assert (ci->constraint_type == DB_CONSTRAINT_UNIQUE || ci->constraint_type == DB_CONSTRAINT_PRIMARY_KEY);
+
+	     error = db_add_constraint (class_mop, ci->constraint_type, NULL, (const char **) ci->att_names, 0);
+	   }
+
+	 if (error != NO_ERROR)
+	   {
+	     goto exit;
+	   }
+       }
+    }
+
+exit:
+
+  if (ctemplate != NULL)
+    {
+      dbt_abort_class (ctemplate);
+      ctemplate = NULL;
+    }
+
+  if (error != NO_ERROR && tran_saved && error != ER_LK_UNILATERALLY_ABORTED)
+    {
+      (void) tran_abort_upto_system_savepoint (UNIQUE_SAVEPOINT_CHANGE_ATTR);
+    }
+
+  if (attr_chg_prop.constr_info != NULL)
```

Comment:
I think that the below 2 conditional codes are not necessary as the above comment.

Reply:
Thank you. The code has been removed at the final commit.